com.unity.robotics.urdf-importer/Runtime/Controller/Controller.cs

```diff
@@ -108,19 +102,16 @@ private void Highlight(int selectedIndex)
             // set the color of the selected join meshes to the highlight color
             foreach (var mesh in rendererList)
             {
-                if (IsHDR())
-                {
-                    mesh.material.SetColor("_BaseColor", highLightColor);
-                }
-                else
-                {
-                    mesh.material.color = highLightColor;
-                }
+                MaterialExtensions.SetMaterialColor(mesh.material, highLightColor);
             }
         }
 
         void DisplaySelectedJoint(int selectedIndex)
         {
+            if (selectedIndex < 0 || selectedIndex >= articulationChain.Length) 
```

Comment:
same. This seems not necessary because of the new method `SetSelectedJointIndex()`?

Reply:
correct. I think this check was added before the SetSelectedJointIndex(). But I think even with that, the index can get out of bounds if URDF is loaded in co routine mode while a game is running